src/server.py

```diff
@@ -344,8 +344,9 @@ class HackSoundServer(Gio.Application):
     """
 
     def __init__(self, metadata):
-        super().__init__(application_id=self._DBUS_NAME,
-                         flags=Gio.ApplicationFlags.IS_SERVICE)
+        Gio.Application.__init__(self, application_id=self._DBUS_NAME,
+                                 flags=Gio.ApplicationFlags.IS_SERVICE)
+        Loggable.__init__(self, ServerFormatter)
```

Comment:
This is a bit awkward since we lose the benefits of `super()` here. Theoretically Python should chain up to the init methods of all the inherited classes if we use `super()` in each one. However, it is extra complicated because the init methods all have different signatures, so it's possible this is still the best way to do it.

There are a few ways worth exploring that we might make it easier. One would be to have the logger be a private member instead of a base class (composition over inheritance.) The other would be to have a dedicated logger class, it only needs to be three lines:
```py
class ServerLogger(Loggable):
    def __init__(self, *args, **kwargs):
        super().__init__(ServerFormatter)

class HackSoundServer(Gio.Application, ServerLogger): ...
```

Reply:
Unfortunately this does not work:
```
In [33]: class HackSoundServer(Y, ServerLogger): 
    ...:     def __init__(self): 
    ...:         super().__init__() 
In [35]: x = HackSoundServer()                                                                                                                       
n [36]: x.info("ters")                                                                                                                              
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-36-36958c43ab58> in <module>
----> 1 x.info("ters")

/usr/lib/python3.7/logging/__init__.py in info(self, msg, *args, **kwargs)
   1329         logger.info("Houston, we have a %s", "interesting problem", exc_info=1)
   1330         """
-> 1331         if self.isEnabledFor(INFO):
   1332             self._log(INFO, msg, args, **kwargs)
   1333 

/usr/lib/python3.7/logging/__init__.py in isEnabledFor(self, level)
   1571         """
   1572         try:
-> 1573             return self._cache[level]
   1574         except KeyError:
   1575             _acquireLock()

AttributeError: 'HackSoundServer' object has no attribute '_cache'


```